standards/DEPENDENCY_UPGRADING.md

```diff
@@ -0,0 +1,93 @@
+Dependency Upgrading Standard
+=============================
+
+## General approach
+
+When working on a repository, developers should check that the dependencies of that repository are sufficiently up to
+date. The approach to updating the dependencies depends on the type of app and the underlying technology.
```

Comment:
I wonder if its worth hammering home the point (either here or in the specific go part) that when it comes to language versions themselves theres no need to wait for anyone to _decide_ we should be updating, that if you notice a new version we should be moving to it (and that it might be worth making others aware in Slack that a new version is available). 

to my mind, dependencies are more obviously the libraries an app is using than they are the language itself, so I think explicitly saying 'we really do mean upgrade the language and you dont have to ask' might be valuable reinforcement. 

Everything you've said is really clearly laid out, and once my brain adjusts it might be superfluous to say it, but theres just a little bit of my brain thinking 'really, I can do that?' and if the documentation said 'YES' I think it might help :)

Reply:
Personally, I would prefer there to be a test that says 'your language version is out of date' than just a policy. With the library dependencies the audit job will make upgrades happen because people can't merge their PR without it. That get's over the upgrading nervousness as it's clear that not doing it is the wrong option. I'd like the same for languages. Eg. when running `make audit` it would fail if the go minor version is not current or previous. I think we should add this as a 20% task.
I will add the words in as well though. 👍

Reply:
I definitely think being nudged is always a step up from having to remember, 20% time candidate for sure :) Thanks for adding the clarification, will approve now!